File: plataforma-felina-backend/src/main/java/com/example/plataforma_felina/domain/Gato.java
package com.example.plataforma_felina.domain;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;

@Entity
@Table(name = "gato")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Gato {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String nombre;

    private String sexo;

    private LocalDate fechaNacimiento;

    private String raza;

    private String color;

    private String vacunas;

    private String caracter;

    private String notas;

    private String estado; //ADOPTABLE, APADRINABLE, ADOPTADO

    private boolean urgente;

    private int edad;

    @Column(name = "foto_url")
    private String fotoUrl;

    @Column(name = "vacunado")
    private Boolean vacunado;

    @Column(name = "castrado")
    private Boolean castrado;

    @Column(name = "desparasitado")
    private Boolean desparasitado;

    @Column(name = "microchip")
    private Boolean microchip;

    @Column(name = "apto_ninos")
    private Boolean aptoNinos;

    @Column(name = "apto_otros_gatos")
    private Boolean aptoOtrosGatos;

    @Column(name = "apto_perros")
    private Boolean aptoPerros;

}

File: plataforma-felina-frontend/src/app/services/gato.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable, BehaviorSubject } from 'rxjs';
import { environment } from '../../environments/environment';



export interface Gato {
  edad: number;
  id: number;
  nombre: string;
  sexo: string;
  fechaNacimiento: string;
  raza: string;
  color: string;
  vacunas: string;
  caracter: string;
  notas: string;
  estado: string | null;
  urgente: boolean; //
  fotoUrl: string;
  vacunado?: boolean;
  castrado?: boolean;
  desparasitado?: boolean;
  microchip?: boolean;
  aptoNinos?: boolean;
  aptoOtrosGatos?: boolean;
  aptoPerros?: boolean;
}

@Injectable({ providedIn: 'root' })
export class GatoService {
  private apiUrl = `${environment.apiUrl}/api/gatos`;

  constructor(private http: HttpClient) {}

  getGatos(): Observable<Gato[]> {
    return this.http.get<Gato[]>(this.apiUrl);
  }

  getGatosDisponibles(): Observable<Gato[]> {
    return this.http.get<Gato[]>(`${this.apiUrl}/disponibles`);
  }

  getGatoById(id: string): Observable<any> {
    return this.http.get<any>(`${this.apiUrl}/${id}`);
  }

  getGatosAdoptados(): Observable<Gato[]> {
    return this.http.get<Gato[]>(`${this.apiUrl}/adoptados`);
  }
}
